pkg/registry/file/vulnsummarystorage_test.go

```diff
@@ -0,0 +1,461 @@
+							All:      20,
+							Relevant: 6,
+						},
+						High: softwarecomposition.VulnerabilityCounters{
+							All:      20,
+							Relevant: 6,
+						},
+						Medium: softwarecomposition.VulnerabilityCounters{
+							All:      20,
+							Relevant: 6,
+						},
+						Low: softwarecomposition.VulnerabilityCounters{
+							All:      20,
+							Relevant: 6,
+						},
+						Negligible: softwarecomposition.VulnerabilityCounters{
+							All:      20,
+							Relevant: 6,
+						},
+						Unknown: softwarecomposition.VulnerabilityCounters{
+							All:      20,
+							Relevant: 6,
+						},
+					},
+					WorkloadVulnerabilitiesObj: []softwarecomposition.VulnerabilitiesObjScope{
+						softwarecomposition.VulnerabilitiesObjScope{
+							Name:      "aaa",
+							Namespace: "bbb",
+							Kind:      "any",
+						},
+						softwarecomposition.VulnerabilitiesObjScope{
+							Name:      "ccc",
+							Namespace: "ddd",
+							Kind:      "many",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run("", func(t *testing.T) {
+			updateFullVulnSumm(tt.fullVulnSumm, tt.vulnManifestSumm)
+			assert.Equal(t, tt.fullVulnSumm.Spec.Severities.Critical.All, tt.expectedFullVulnSumm.Spec.Severities.Critical.All)
+			assert.Equal(t, tt.fullVulnSumm.Spec.Severities.Critical.Relevant, tt.expectedFullVulnSumm.Spec.Severities.Critical.Relevant)
+
+			assert.Equal(t, tt.fullVulnSumm.Spec.Severities.High.All, tt.expectedFullVulnSumm.Spec.Severities.High.All)
+			assert.Equal(t, tt.fullVulnSumm.Spec.Severities.High.Relevant, tt.expectedFullVulnSumm.Spec.Severities.High.Relevant)
+
+			assert.Equal(t, tt.fullVulnSumm.Spec.Severities.Medium.All, tt.expectedFullVulnSumm.Spec.Severities.Medium.All)
+			assert.Equal(t, tt.fullVulnSumm.Spec.Severities.Medium.Relevant, tt.expectedFullVulnSumm.Spec.Severities.Medium.Relevant)
+
+			assert.Equal(t, tt.fullVulnSumm.Spec.Severities.Low.All, tt.expectedFullVulnSumm.Spec.Severities.Low.All)
+			assert.Equal(t, tt.fullVulnSumm.Spec.Severities.Low.Relevant, tt.expectedFullVulnSumm.Spec.Severities.Low.Relevant)
+
+			assert.Equal(t, tt.fullVulnSumm.Spec.Severities.Negligible.All, tt.expectedFullVulnSumm.Spec.Severities.Negligible.All)
+			assert.Equal(t, tt.fullVulnSumm.Spec.Severities.Negligible.Relevant, tt.expectedFullVulnSumm.Spec.Severities.Negligible.Relevant)
+
+			assert.Equal(t, tt.fullVulnSumm.Spec.Severities.Unknown.All, tt.expectedFullVulnSumm.Spec.Severities.Unknown.All)
+			assert.Equal(t, tt.fullVulnSumm.Spec.Severities.Unknown.Relevant, tt.expectedFullVulnSumm.Spec.Severities.Unknown.Relevant)
+
+			assert.Equal(t, len(tt.fullVulnSumm.Spec.WorkloadVulnerabilitiesObj), len(tt.expectedFullVulnSumm.Spec.WorkloadVulnerabilitiesObj))
+			for i := range tt.fullVulnSumm.Spec.WorkloadVulnerabilitiesObj {
+				assert.Equal(t, tt.fullVulnSumm.Spec.WorkloadVulnerabilitiesObj[i], tt.expectedFullVulnSumm.Spec.WorkloadVulnerabilitiesObj[i])
+			}
```

Comment:
question: why make all these asserts separately when you provide an entire expected object as a test input?

Wouldn’t it be more convenient to just compare the objects like:
```
got := FuncBeingTested()
assert.Equal(t, tc.expectedFullVulnSumm, got)
```